pkg/kubectl/rolebinding.go

```diff
@@ -147,13 +149,17 @@ func (s RoleBindingGeneratorV1) StructuredGenerate() (runtime.Object, error) {
 	}
 	for _, sa := range sets.NewString(s.ServiceAccounts...).List() {
 		tokens := strings.Split(sa, ":")
-		if len(tokens) != 2 {
+		if len(tokens) != 2 || tokens[1] == "" {
 			return nil, fmt.Errorf("serviceaccount must be <namespace>:<name>")
 		}
+		namespace := tokens[0]
+		if len(namespace) == 0 {
+			namespace = s.DefaultNamespace
```

Comment:
I don't think we need to default this value.  It's allowed to be empty and the authorizer will accept the empty and have it mean, "namespace of the rolebinding".  This allows for `kubectl get -o yaml | kubectl create -f - -n different` to work how you'd expect.